kafka-rest/src/main/java/io/confluent/kafkarest/controllers/ConsumerGroupLagManagerImpl.java

```diff
@@ -0,0 +1,118 @@
+import io.confluent.kafkarest.entities.ConsumerGroupLag;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import javax.inject.Inject;
+import javax.ws.rs.NotFoundException;
+import org.apache.kafka.clients.admin.Admin;
+import org.apache.kafka.clients.admin.ListOffsetsResult.ListOffsetsResultInfo;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class ConsumerGroupLagManagerImpl extends AbstractConsumerLagManager implements ConsumerGroupLagManager {
+
+  private final ConsumerGroupManager consumerGroupManager;
+  private static final Logger log = LoggerFactory.getLogger(ConsumerGroupLagManagerImpl.class);
+
+  @Inject
+  ConsumerGroupLagManagerImpl(
+      Admin kafkaAdminClient,
+      ConsumerGroupManager consumerGroupManager) {
+    super(kafkaAdminClient);
+    this.consumerGroupManager = requireNonNull(consumerGroupManager);
+  }
+
+  @Override
+  public CompletableFuture<Optional<ConsumerGroupLag>> getConsumerGroupLag(
+      String clusterId,
+      String consumerGroupId
+  ) {
+    return consumerGroupManager.getConsumerGroup(clusterId, consumerGroupId)
+        .thenApply(
+            consumerGroup ->
+                checkEntityExists(consumerGroup, "Consumer Group %s could not be found.", consumerGroupId))
+        .thenCompose(
+            consumerGroup ->
+                getCurrentOffsets(consumerGroupId)
+                    .thenApply(
+                        fetchedCurrentOffsets ->
+                            checkOffsetsExist(fetchedCurrentOffsets, "Consumer group offsets could not be found."))
+                    .thenCompose(
+                        fetchedCurrentOffsets ->
+                            getLatestOffsets(fetchedCurrentOffsets)
+                                .thenApply(
+                                    latestOffsets ->
+                                       createConsumerGroupLag(
+                                           clusterId,
+                                           consumerGroup,
+                                           fetchedCurrentOffsets,
+                                           latestOffsets))));
+  }
+
+  Optional<ConsumerGroupLag> createConsumerGroupLag(
+      String clusterId,
+      ConsumerGroup consumerGroup,
+      Map<TopicPartition, OffsetAndMetadata> fetchedCurrentOffsets,
+      Map<TopicPartition, ListOffsetsResultInfo> latestOffsets) {
+    Map<TopicPartition, MemberId> tpMemberIds =
+        getMemberIds(consumerGroup);
+    ConsumerGroupLag.Builder consumerGroupLag =
+        ConsumerGroupLag.builder()
+            .setClusterId(clusterId)
+            .setConsumerGroupId(consumerGroup.getConsumerGroupId());
+    fetchedCurrentOffsets.keySet().stream().forEach(
+        topicPartition -> {
+          MemberId memberId = tpMemberIds.getOrDefault(
```

Comment:
Instead of creating a dummy here, you should keep this optional, and use `map` below, where you use it. E.g.:


```
Optional<MemberId> memberId = Optional.ofNullable(tpMemberIds.get(topicPartition));
...
consumerGroupLag.addOffset(
    topicPartition.topic(),
    memberId.map(MemberId::getConsumerId).orElse(""),
    memberId.map(MemberId::getClientId).orElse(""),
    memberId.map(MemberId::getInstanceId).orElse(null),
    topicPartition.partition(),
    currentOffset,
    latestOffset);
```